pkg/monitor/cluster/cluster.go

```diff
@@ -100,6 +100,13 @@ func NewMonitor(ctx context.Context, log *logrus.Entry, restConfig *rest.Config,
 func (mon *Monitor) Monitor(ctx context.Context) (errs []error) {
 	mon.log.Debug("monitoring")
 
+	if mon.hourlyRun {
+		mon.emitGauge("cluster.provisioning", 1, map[string]string{
+			"provisioningState":       mon.oc.Properties.ProvisioningState.String(),
+			"failedProvisioningState": mon.oc.Properties.FailedProvisioningState.String(),
```

Comment:
For my own understanding, what wis the value of the `provisioningState` dimension, if a cluster failed to be created? I assume the `failedProvisioningState` dimension will be set to `Creating`, and the value of this metric will be `1`? If the cluster recovers in the next hourly run, the dimension values will be reversed ? Will the dashboard use these dimensions to show the recovery?

Reply:
> For my own understanding, what is the value of the provisioningState dimension, if a cluster failed to be created? I assume the failedProvisioningState dimension will be set to Creating, and the value of this metric will be 1?

Any cluster in the RP DB will emit 1 with dimension containing combination of `provisioningState` and `failedProvisioningState`.

`provisioningState` can either be non-terminal (e.g. `Creating`, `Updating`, `Deleting` or `AdminUpdating`) or terminal (eg. `Succeeded`, `Failed`). When you start cluster creation process, `provisioningState` going to be `Creating` and `failedProvisioningState` empty. This metric will emit 1 with this combination.

Once the cluster fails to create `provisioningState` wil be `Failed` and `failedProvisioningState` will contain `Creating`. This metric will emit 1 with this combination.

> If the cluster recovers in the next hourly run, the provisioningState dimension values will be Succeeded ?

We don't recover cluster from the `Failed` `Creating` and `Deleting` state at the moment. This story comes from the discussion about the need to do some clean ups for `Failed`/`Deleting` clusters. Before we take any action, we want to gather some data. We want to see:

1. How many clusters we have in the failed states (including `Failed`/`Deleting`)
2. Do our cganges (such as deleting VMs before other resoureces) help us to reduce the rate of `Failed`/`Deleting`.
3. Not relevant to the clean up, but the dashbaord with these dimensions can be useful to see trends across all clusters. It should be able to show unsually quickly increasning number of failed clsuters and spikes of failed clsuters.

> Will the dashboard use these dimensions to show the recovery?

Related to the above. You should be able to filter by failed state (or specific combination) and see whether the number goes up or down. But it is not necessary gonig to be recovery: `Failed`/`Creating` can be deleted by the custoemr and the number will go down too.

It is basically a summary of what we had in the DB over the time.

Reply:
Thanks for the explanation!